src/charm.py

```diff
@@ -147,7 +147,11 @@ def _on_leader_elected(self, _: ops.LeaderElectedEvent) -> None:
         if not container.can_connect():
             self.unit.status = ops.MaintenanceStatus("Waiting for pebble")
             return
-        admin_user = actions.register_user(container, SYNAPSE_OPERATOR_USER, True)
+        # Using 16 to create a random value but to  be secure against brute-force attacks, please
+        # check the docs:
+        # https://docs.python.org/3/library/secrets.html#how-many-bytes-should-tokens-use
+        username = token_hex(16)
+        admin_user = actions.register_user(container, username, True)
```

Comment:
Why a random username?

Reply:
Kind of a egg-chicken problem.
If the user exists, register_user will try to get the access_token. To do that it needs an admin user. So, to be on the safe side, I decided to use a random user name. So, if for any reason there is no access token on peer data/secret, another user will be created.

Reply:
Thanks for the explanation :) Would you mind adding a comment exaplining why? I'm a bit concerned about someone removing this because they lack the reasoning behind it

Reply:
Sure, thanks. Added.